### PR TITLE
three spottypes added to models, migrated and tested

### PR DESCRIPTION
**Pull request by omaraljazairy**

### CODE REVIEW

File: src/datasets/blackspots/models.py [hunk 1]
```diff
@@ -19,6 +19,10 @@ class SpotType(DjangoChoices):
         protocol_ernstig = ChoiceItem()
         protocol_dodelijk = ChoiceItem()
         risico = ChoiceItem()
+        gebiedslocatie_ivm = ChoiceItem()
```
Comment (jklardie) [inline src/datasets/blackspots/models.py:22]: add a comment to explain what ivm is ("infrastructurele verkeersveiligheidsmaatregel")

File: src/datasets/blackspots/models.py [hunk 2]
```diff
@@ -19,6 +19,10 @@ class SpotType(DjangoChoices):
...
+        schoolstraat = ChoiceItem()
+        vso = ChoiceItem()
```
Comment (jklardie) [inline src/datasets/blackspots/models.py:24]: add a comment to explain what vso is ("veilige schoolomgeving")

File: tests/api/test_api.py [hunk 3]
```diff
@@ -125,6 +125,66 @@ def test_spot_detail_post(self, determine_stadsdeel):
...
+            "point": '{"type": "Point","coordinates": [4.9239023,52.3875655]}',
+            "actiehouders": "Actiehouders test vso",
+            "status": "voorbereiding",
+        }
+        response = self.write_client.post(url, data=data)
+        self.assertStatusCode(url, response, expected_status=201)
+
+        del data["point"]
+        self.assertTrue(Spot.objects.filter(**data).exists())
+
```
Comment (jklardie) [inline tests/api/test_api.py:187]: These testcases (and the one that already existed, `test_spot_detail_post`) are practically the same code. You should follow the DRY principle. In this case I do want to test all possible spot types, but the code itself should be a single function. 

To do so you can use parametrization. See https://docs.pytest.org/en/7.1.x/how-to/parametrize.html

**Review by jklardie — CHANGES_REQUESTED** [review]

File: tests/api/test_api.py (hunk 3 above)
Comment (omaraljazairy) [reply tests/api/test_api.py]: The paramertrize decorator of pytest doesn't work with unittest. I tried it and it didn't work and then I saw that according to https://pytest.org/en/7.1.x/how-to/unittest.html this is not supported. There are many solutions, like using an external library parameterized, or creating a separate class that doesn't inherit from TestCase. But the cleanest one I saw was just create a list of tuple of objets and loop through it, the same what the decorator does at the end. If you suggest a better solution, let me please know.

File: tests/api/test_api.py [hunk 4]
```diff
@@ -105,25 +105,67 @@ def test_spot_detail_get_auth_error(self):
...
-            "status": "voorbereiding",
-            "jaar_blackspotlijst": 2019,
-        }
-        response = self.write_client.post(url, data=data)
-        self.assertStatusCode(url, response, expected_status=201)
+        #  the @pytest.mark.parametrize is not supported with unittests. The 
+        #  easiest and cleanest way is so create a list
+        #  of tuples and loop through it. The same what the parametrize 
+        #  decorator does. the tuple contains the data object but can also
+        #  contain the status code and other variables.
```
Comment (jklardie) [inline tests/api/test_api.py:117]: Ook dit vind ik geen geschikte oplossing. Sowieso plaats je nu de dictionary met data in een set en die in een array. De set is overbodig. Maar daarnaast is dit nog steeds niet DRY. Het grootste deel van deze array overlapt. Dan heb ik nog liever dat we parametrize namaken op deze manier:

```
    def test_spot_detail_post(self):
        for spot_type, _label in Spot.SpotType.choices:
            self._test_spot_detail_post(spot_type=spot_type)

    @mock.patch("api.serializers.SpotSerializer.determine_stadsdeel")
    def _test_spot_detail_post(self, determine_stadsdeel, spot_type):
        determine_stadsdeel.return_value = Spot.Stadsdelen.Centrum

        url = reverse("spot-list")
        data = {
            "locatie_id": random.randint(1, 999999),
            "spot_type": spot_type,
            "description": "Test spot",
            "point": '{"type": "Point","coordinates": [4.9239022,52.3875654]}',
            "actiehouders": "Actiehouders test",
            "status": "voorbereiding",
        }

        if spot_type in [Spot.SpotType.blackspot, Spot.SpotType.wegvak]:
            data["jaar_blackspotlijst"] = 2019
        elif spot_type in [Spot.SpotType.protocol_ernstig, Spot.SpotType.protocol_dodelijk]:
            data["jaar_ongeval_quickscan"] = 2020

        response = self.write_client.post(url, data=data)
        self.assertStatusCode(url, response, expected_status=201)

        del data["point"]
        self.assertTrue(Spot.objects.filter(**data).exists())
```

**Review by jklardie — CHANGES_REQUESTED** [review]

**Pull request merged**